reccmp/tools/datacmp.py

```diff
@@ -142,6 +142,7 @@ def create_comparison_item(
 
 
 def do_the_comparison(target: RecCmpBuiltTarget) -> Iterable[ComparisonItem]:
+    # pylint: disable=too-many-locals
```

Comment:
Optional: Is there a way to reasonably split this function into multiple ones? It is quite long, which is likely related to why we need this suppression. No need to refactor this before merging, but if you feel like it, give it a shot.